File: services/upx/Entities/LocationClaim.cs
namespace Foundation.Services.UPx;

public class LocationClaim
{
    public float Latitude { get; set; } = default!;

    public float Longitude { get; set; } = default!;

    public long Timestamp { get; set; } = default!;

    public TimeSpan Age => DateTime.Now - DateTimeOffset.FromUnixTimeSeconds(Timestamp).DateTime;

    public string StationId { get; set; } = default!;
}

File: services/echo/internal/structures/location_claim.go
package structures

import "time"

type LocationClaim struct {
	Latitude  float32       `json:"latitude"`
	Longitude float32       `json:"longitude"`
	Timestamp int64         `json:"timestamp"`
	StationId string        `json:"station_id"`
	Age       time.Duration `json:"age"`
}
